packages/tools/addon/templates/components/cardstack-tools.hbs

```diff
@@ -1,18 +1,16 @@
-{{#toolbar-manager topRightCornerBelongsTo="right"}}
+{{#cardstack-edges topRightCornerBelongsTo="right" as |edges|}}
   {{yield}}
 
-  {{#fixed-within-toolbars top=0 right=0}}
+  {{#edges.launcher}}
     {{cardstack-tools-launcher}}
-  {{/fixed-within-toolbars}}
-
-{{/toolbar-manager}}
+  {{/edges.launcher}}
+{{/cardstack-edges}}
 
 {{#if tools.active}}
-  {{in-top-toolbar show=(component "cardstack-header") }}
-  {{in-right-toolbar show=(component "cardstack-toolbox") }}
+  {{cardstack-edge 'top' (component "cardstack-header")}}
+  {{cardstack-edge 'right' (component "cardstack-toolbox")}}
   {{cs-field-overlays}}
```

Comment:
The goal as you keep refactoring will be to delete the cardstack-tools component.

We want any number of optional packages to be able to inject stuff like this:

```hbs
{{#edges.launcher}}
     {{cardstack-tools-launcher}}
{{/edges.launcher}}

{{#if tools.active}}
  {{cardstack-edge 'top' (component "cardstack-header")}}
  {{cardstack-edge 'right' (component "cardstack-toolbox")}}
  {{cs-field-overlays}}
{{/if}}

{{from-elsewhere name="paragraph-tools"}}
```

into the app's top-level template. Probably in the implementation of cardstack-edges it could be something like:

```hbs
{{#each registeredServices as |s|}}
  {{#if s.injectComponent}}
      {{component s.injectComponent edges=edges}}
  {{/if}}
{{/each}}
```